src/test/java/org/broadinstitute/hellbender/utils/pileup/ReadPileupUnitTest.java

```diff
@@ -490,4 +491,106 @@ public void testSimplePileupWithOffset(){
+    }
+
+    @Test
+    public void testFixPairOverlappingQualitiesCap() {
+        final PileupElement element1 = PileupElement
+                .createPileupForReadAndOffset(ArtificialReadUtils.createArtificialRead("1M"), 0);
+        element1.getRead().setName("read1");
+        element1.getRead().setBases(new byte[] {'A'});
+        final PileupElement element2 = PileupElement
+                .createPileupForReadAndOffset(ArtificialReadUtils.createArtificialRead("1M"), 0);
+        element2.getRead().setName("read2");
+        element2.getRead().setBases(new byte[] {'A'});
+        final PileupElement element3 = PileupElement
+                .createPileupForReadAndOffset(ArtificialReadUtils.createArtificialRead("1M"), 0);
+        element3.getRead().setName("read3");
+        element2.getRead().setBases(new byte[] {'A'});
+        element3.getRead().setBaseQualities(new byte[] {Byte.MAX_VALUE});
+        // check all possible combinations that goes beyond the maximum value
+        for (byte i = 0; i < Byte.MAX_VALUE; i++) {
+            final byte[] iArray = new byte[] {i};
+            final byte j = (byte) (Byte.MAX_VALUE - i);
+            element1.getRead().setBaseQualities(iArray);
+            element2.getRead().setBaseQualities(new byte[] {j});
+            logger.debug("Test: fixing ({}) and ({})", element1, element2);
+            ReadPileup.fixPairOverlappingQualities(element1, element2);
+            Assert.assertEquals(element1.getQual(), Byte.MAX_VALUE);
+            Assert.assertEquals(element2.getQual(), 0);
+            element1.getRead().setBaseQualities(iArray);
+            logger.debug("Test: fixing ({}) and ({})", element3, element1);
+            ReadPileup.fixPairOverlappingQualities(element3, element1);
+            Assert.assertEquals(element3.getQual(), Byte.MAX_VALUE);
+            Assert.assertEquals(element1.getQual(), 0);
+        }
+        // finally, check what happen if both are the maximum value
+        element1.getRead().setBaseQualities(new byte[] {Byte.MAX_VALUE});
+        logger.debug("Test: fixing ({}) and ({})", element3, element1);
+        ReadPileup.fixPairOverlappingQualities(element3, element1);
+        Assert.assertEquals(element3.getQual(), Byte.MAX_VALUE);
+        Assert.assertEquals(element1.getQual(), 0);
+    }
+
+    @Test
+    public void testFixOverlappingQualitiesDeletedElements() throws Exception {
+        final GATKRead read = ArtificialReadUtils.createArtificialRead("1M1D1M");
+        read.setBases(new byte[] {'A', 'A'});
+        read.setBaseQualities(new byte[] {60, 60});
+        final GATKRead read2 = ArtificialReadUtils.createArtificialRead("1M1D1M");
+        read.setBases(new byte[] {'A', 'A'});
+        read2.setBaseQualities(new byte[] {40, 40});
+        final GATKRead copy = read.deepCopy();
+        final GATKRead copy2 = read2.deepCopy();
+        final PileupElement deleted = new PileupElement(copy, 1, copy.getCigarElement(1), 1, 0);
```

Comment:
I will add a comment, but just for clarify it here too: because the method `ReadPileup.fixPairOverlappingQualities()` modify the read inside the `PileupElement`, we need a way to check that anything within the read was changed. Thus, we require a copy to maintain a backup of the original read and check if it was change or not. Is this clear like that, because I guess that is still a bit confusing?